File: Egg/solution/sol.cpp
#include "Egg.h"

long long height_limit(long long M) {
	long long l = 1, r = M + 1;
	while (r - l > 1) {
		long long mid = (l + r) >> 1;
		if (is_broken(mid)) r = mid;
		else l = mid;
	}
	return l;
}

File: Egg/solution/sol.py
def height_limit(M: int) -> int:
    from __main__ import is_broken
    l, r = 1, M + 1
    while r - l > 1:
        mid = (l + r) // 2
        if is_broken(mid) == 1:
            r = mid
        else:
            l = mid
    return l
